drivers/misc/optee.c

```diff
@@ -1075,6 +1075,7 @@ optee_ioctl_shm_alloc(FAR struct tee_ioctl_shm_alloc_data *data)
   return get_errno();
 }
 
+#ifdef CONFIG_DEV_OPTEE_SMC
```

Comment:
let's remove the check, we are planing to support the shared memory for rpmsg/local socket too